common/djangoapps/student/tests/test_reset_password.py

```diff
@@ -252,6 +252,31 @@ def test_reset_password_good_token(self):
         self.user = User.objects.get(pk=self.user.pk)
         self.assertTrue(self.user.is_active)
 
+    def test_non_matching_password_reset(self):
+        """
+        Tests that if we providing mismatching passwords
+
+        Given user is not active when we try to submit confirm password rest form, User is
+        provided helpful error message.
+        """
+        # Given that user is not active.
+        self.assertFalse(self.user.is_active)
+        url = reverse(
+            'password_reset_confirm',
+            kwargs={'uidb36': self.uidb36, 'token': self.token}
+        )
+        request_params = {'new_password1': 'password1', 'new_password2': 'password2'}
+        confirm_request = self.request_factory.post(url, data=request_params)
+
+        # Make a request and verify the error message in context.
+        confirm_request_resp = password_reset_confirm_wrapper(confirm_request, self.uidb36, self.token)
```

Comment:
Note that err_msg is also tested below in test_non_matching_password_reset (in the other test file). I wouldn't recommend duplicating the coverage. It looks like this test is only interested in whether or not the user is active (like test_reset_password_bad_token).